clippy_lints/src/utils/author.rs

```diff
@@ -333,6 +334,24 @@ impl<'a, 'tcx> PrintVisitor<'a, 'tcx> {
 
     #[allow(clippy::too_many_lines)]
     fn expr(&self, expr: &Binding<&hir::Expr<'_>>) {
+        if let Some(macro_call) = root_macro_call_first_node(self.cx, expr.value)
+            && is_format_macro(self.cx, macro_call.def_id)
+        {
+            let diag = self.cx.tcx.item_name(macro_call.def_id);
+            bind!(self, macro_call);
+            chain!(self, "let Some({macro_call}) = macros::root_macro_call_first_node(cx, {expr})");
+            chain!(self, "cx.tcx.is_diagnostic_item(sym::{diag}_macro, {macro_call}.def_id)");
+            let exprs: Vec<_> = collect_format_args(expr.value).collect();
+            let args = exprs.as_slice();
+            bind!(self, args);
+            chain!(self, "let {args} = macros::collect_format_args({expr}).collect()");
+            self.slice(args, |e| {
+                let expr = Binding { name: e.name.clone(), value: *e.value };
```

Comment:
I really don't like this `clone`, but I'm not really sure how to avoid it here...

Reply:
I've changed the signature of `slice` to provide an owned `Binding` as it already has it, not sure why it was by-ref previously, but that solved the issue of unnecessary clone.